AK/JsonParser.cpp

```diff
@@ -310,17 +350,33 @@ JsonValue JsonParser::parse()
     case '7':
     case '8':
     case '9':
-        return parse_number();
+        return parse_number(ok);
     case 'f':
-        return parse_false();
+        return parse_false(ok);
     case 't':
-        return parse_true();
+        return parse_true(ok);
     case 'n':
-        return parse_null();
-    case 'u':
-        return parse_undefined();
+        return parse_null(ok);
     }
 
+    ok = false;
     return JsonValue();
 }
+
+JsonValue JsonParser::parse()
+{
+    bool ok = true;
+    auto result = parse(ok);
+    ASSERT(ok);
+    return result;
+}
+
+JsonValue JsonParser::parse(bool& ok) {
```

Comment:
Please use `clang-format`.
Also please make this kind of API return `Optional<JsonValue>` instead of using an out-parameter. :)